include/mbedtls/threading.h

```diff
@@ -70,20 +126,27 @@ typedef struct
  * \param mutex_lock    the lock function implementation
  * \param mutex_unlock  the unlock function implementation
  */
-void mbedtls_threading_set_alt( void (*mutex_init)( mbedtls_threading_mutex_t * ),
-                       void (*mutex_free)( mbedtls_threading_mutex_t * ),
-                       int (*mutex_lock)( mbedtls_threading_mutex_t * ),
-                       int (*mutex_unlock)( mbedtls_threading_mutex_t * ) );
+void mbedtls_threading_set_alt( mbedtls_mutex_init_t* mutex_init,
+                                mbedtls_mutex_free_t* mutex_free,
+                                mbedtls_mutex_lock_t* mutex_lock,
+                                mbedtls_mutex_unlock_t* mutex_unlock );
 
 /**
  * \brief               Free global mutexes.
  */
 void mbedtls_threading_free_alt( void );
-#endif /* MBEDTLS_THREADING_ALT */
+
+#else
+/* For no given implementation, define as an opaque pointer */
+typedef struct _mbedtls_threading_mutex_t
+{
+    void* mutex;
+} mbedtls_threading_mutex_t;
+#endif
```

Comment:
Could you please add a comment next to the `#endif` with the condition of the matching `#if`?